import "mapreduce/yt/interface/protos/extension.proto";

package NAlice.NUniproxy;

message TVoiceInput {
    optional string RequestId = 1 [(NYT.key_column_name) = "RequestId"];
    optional string Uuid = 2 [(NYT.key_column_name) = "Uuid"];
    optional string Topic = 3 [(NYT.column_name) = "Topic"];
    optional string ApplicationId = 4 [(NYT.column_name) = "ApplicationId"];
    optional string ApplicationVersion = 5 [(NYT.column_name) = "ApplicationVersion"];
    optional string OsVersion = 6 [(NYT.column_name) = "OsVersion"];
    optional string Platform = 7 [(NYT.column_name) = "Platform"];
    optional string Timezone = 8 [(NYT.column_name) = "Timezone"];
    optional string Lang = 9 [(NYT.column_name) = "Lang"];
    optional string Format = 10 [(NYT.column_name) = "Format"];
    optional string Payload = 11 [(NYT.column_name) = "Payload"];
    optional bytes VoiceData = 12 [(NYT.column_name) = "VoiceData"];
    optional string AuthToken = 13 [(NYT.column_name) = "AuthToken"];
    optional string TextData = 14 [(NYT.column_name) = "Text"];
    optional string FetcherMode = 15 [(NYT.column_name) = "FetcherMode"];
    optional string SyncStateExperiments = 16 [(NYT.column_name) = "SyncStateExperiments"];
    optional string OAuthToken = 17 [(NYT.column_name) = "OAuthToken"];
    optional string Application = 18 [(NYT.column_name) = "Application"];
    optional string ClientTime = 19 [(NYT.column_name) = "ClientTime"];
    optional bool DisableLocalExperiments = 20 [(NYT.column_name) = "DisableLocalExperiments"];
    optional string SessionId = 21 [(NYT.column_name) = "SessionId"];
    optional int64 SessionSequence = 22 [(NYT.column_name) = "SessionSequence"];
    optional string Headers = 23 [(NYT.column_name) = "Headers"];
    optional string SetraceUrl = 24 [(NYT.column_name) = "SetraceUrl"];
    optional string ShootingSource = 25 [(NYT.column_name) = "ShootingSource"];
    optional string SupportedFeatures = 26 [(NYT.column_name) = "SupportedFeatures"];
    optional int64 ReversedSessionSequence = 27 [(NYT.column_name) = "ReversedSessionSequence"];
};
